### DISCUSSION TITLE
How to get OpenTelemetry traceId in code?

### DISCUSSION THREAD

**Discussion by NMichas** [quarkusio/quarkus#27125] [Q&A] [answered]
Hello,

I'm trying to get a reference in my code to the current context's traceId. I get a reference to the context with:
`Context context = Context.current();`

I can see in the debugger that the above `context` contains the information I need:
![Cursor_and_esthesis-iot_–_QErrorHandler_java__srv-crypto-impl_](https://user-images.githubusercontent.com/2218296/182776862-39fa7364-7905-4a7b-9062-723d7bf5daed.png)

But I don't know how I can extract the `traceId` from it. I tried:
```
ContextKey<ImmutableSpanContext> key = ContextKey.named("opentelemetry-trace-span-key");
System.out.println(context.get(key));
```
...but doesn't work (returns `null`).

Any ideas please? :)

[a comment by an automated account is not shown]

**Reply by radcortez** [reply]
Try `Span.current().getSpanContext().getTraceId()`

**Reply by NMichas** [reply]
Thanks a lot, that worked!

**Reply by gt-logistiex** [reply]
thanks, this worked for me 👍🏻

**Comment by AayushSaini101** [comment]
> Span.current().getSpanContext().getTraceId() @NMichas @gt-logistiex @radcortez 
Can you tell which dependency you are using. I cannot get getTraceId() option: 

<img width="699" alt="image" src="https://github.com/quarkusio/quarkus/assets/60972989/e026a7af-fb92-40f5-8186-1e216b23c403">

**Reply by brunobat** [reply]
I have no Idea of what Span are you using or what version of OpenTelemetry. The solution from Roberto works: ` String traceId = Span.current().getSpanContext().getTraceId();`

**Comment by NMichas** [comment]
`io.opentelemetry.api.trace.Span`

**Reply by AayushSaini101** [reply]
@NMichas I am getting  for both traceID and spanID. I am using GraphQL and SpringBoot
00000000000000000000000000000000
0000000000000000

**Reply by brunobat** [reply]
@AayushSaini101 Are you aware this is a Quarkus discussion forum, not SpringBoot, right?
Saying that, The zeros show up in the parent trace id and span id when there is no parent.

**Comment by NMichas** [comment]
Check if you have a valid context, `Context.current()`. If not, there's something wrong in your setup, I'd start with a stripped down example that works and build up from there.

**Reply by AayushSaini101** [reply]
@NMichas Yes, The Context.current is returning empty entries
<img width="699" alt="image" src="https://github.com/quarkusio/quarkus/assets/60972989/6ad8b71c-8655-4b19-b581-c2a466c41cf7">

**Reply by gt-logistiex** [reply]
did you do this?
import io.opentelemetry.api.trace.Span;

**Reply by AayushSaini101** [reply]
Thanks for the reply @gt-logistiex , Yes i have done
